src/legacy/ui/public/url/url.js

```diff
@@ -47,6 +50,31 @@ export function KbnUrlProvider($injector, $location, $rootScope, $parse, Private
    */
   const self = this;
 
+  /**
+   * Override the behavior of angular's $location.url() to encode/decode
+   * URL parameters
+   * @param {String} url
+   * @returns $location
+   */
+  $location.url = url => {
```

Comment:
I added a breakpoint here, and it seems that `url` has already been decoded at this point.

Going to `http://localhost:5601/app/apm#/opbeans-node/transactions/request/GET%20%2Fapi%2Fcustomers`, I get `url="/opbeans-node/transactions/request/GET%20/api/customers"`

Reply:
Thanks for testing! I'll take a look and get this fixed.

Reply:
Found it:
https://github.com/elastic/kibana/blob/7198a1ffcd58a79b13f568fbb8d9e0a88d486690/src/legacy/ui/public/chrome/directives/kbn_chrome.js#L77-L88

Specifically:
https://github.com/elastic/kibana/blob/7198a1ffcd58a79b13f568fbb8d9e0a88d486690/src/legacy/ui/public/chrome/directives/kbn_chrome.js#L80

`newUrl` is correct but `absUrlHash` decoded the forward slashes. 

@spalger you might find this interesting/infuriating :p

Reply:
Btw even if you don't have any APM data, you should be able to repro this issue by going to the APM app:
`/app/apm#/any-service/transactions/request/GET%20/api/customers`

Reply:
Should these two "corrections" be merged into a single thing?

Reply:
@joshdover I'm not following?

Reply:
oh, whether stuff in `ui/public/chrome/directives/kbn_chrome.js` and `ui/public/url/url.js` should be merged? Yes, if possible, that sounds like a good improvement 👍